File: BEECROWD/1022-TDA-Racional/Main.java
import java.util.Scanner;

public class Main {

	public static void valorRacional(int N1, int D1, String c, String operador, int N2, int D2) {
		int totalN, 
		    totalD,
		    res = N1*D1;
		char soma = '+';
		char subtracao = '-';
		char multiplicacao = '*';
		char divisao = '/';

		if (operador.charAt(0) == soma) {
			totalN = (N1*D2 + N2*D1);
			totalD = (D1*D2);
			System.out.println(totalN + c + totalD + " = " + (totalN/res) + c + (totalD/res));
		}
		else if (operador.charAt(0) == subtracao) {
			totalN = (N1*D2 - N2*D1);
			totalD = (D1*D2);
			System.out.println(totalN + c + totalD + " = " + (totalN/res) + c + (totalD/res));
		}
		else if (operador.charAt(0) == multiplicacao) {
			totalN = (N1*N2);
			totalD = (D1*D2);
			System.out.println(totalN + c + totalD + " = " + (totalN/res) + c + (totalD/res));
		}
		else if (operador.charAt(0) == divisao) {
			totalN = (N1*D2);
			totalD = (N2*D1);
			System.out.println(totalN + c + totalD + " = " + (totalN/res) + c + (totalD/res));
		}
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		int n = sc.nextInt();
		int i = 0,
		    N1,
		    D1,
		    N2,
		    D2;
		String c,
		     operador;

		while(i < n) {
			N1 = sc.nextInt();
			c = sc.next();
			D1 = sc.nextInt();

			operador = sc.next();

			N2 = sc.nextInt();
			c = sc.next();
			D2 = sc.nextInt();
			sc.nextLine();

			valorRacional(N1, D1, c, operador, N2, D2);
			i++;
		}

		sc.close();
	}
}

File: BEECROWD/1022-TDA-Racional/main.c
#include <stdio.h>
#include <stdlib.h>

void TDARacional(int N1, int D1, int N2, int D2, char c, char operador) {
	int totalN,
	    totalD,
	    dividir = N1*D1;

	if (operador == '+') {
		totalN = (N1*D2 + N2*D1);
		totalD = (D1*D2);
		printf("%d%c%d = %d%c%d\n", totalN, c, totalD, (totalN/dividir), c, (totalD/dividir));
	}
	else if (operador == '-') {
		totalN = (N1*D2 - N2*D1);
		totalD = (D1*D2);
		printf("%d%c%d = %d%c%d\n", totalN, c, totalD, (totalN/dividir), c, (totalD/dividir));
	
	}
	else if (operador == '*') {
		totalN = (N1*N2);
		totalD = (D1*D2);
		printf("%d%c%d = %d%c%d\n", totalN, c, totalD, (totalN/dividir), c, (totalD/dividir));

	}
	else if (operador == '/') {
		totalN = (N1*D2);
		totalD = (N2*D1);
		printf("%d%c%d = %d%c%d\n", totalN, c, totalD, (totalN/dividir), c, (totalD/dividir));
	}
}

int main() {
	int N = 0,
	    N1 = 0,
	    N2 = 0,
	    D1 = 0,
	    D2 = 0;
	char operador,
	     fracao;

	scanf("%d", &N);

	for (int i = 0; i < N; i++) {
		scanf("%d", &N1);
		scanf(" %c", &fracao);
		scanf("%d", &D1);

		scanf(" %c", &operador);

		scanf("%d", &N2);
		scanf(" %c", &fracao);
		scanf("%d", &D2);

		TDARacional(N1, D1, N2, D2, fracao, operador);
	}

	return 0;
}
